src/module.rs

```diff
@@ -1169,3 +1359,127 @@ impl Default for Alignments {
+
+    #[test]
+    fn test_size_in_bits() {
+        let dl = DataLayout::default();
+        let types = TypesBuilder::new().build();
+        assert_eq!(
+            Some(TypeSize::fixed(1)),
+            dl.get_type_size_in_bits(&types, &types.bool())
+        );
+        assert_eq!(
+            Some(TypeSize::fixed(8)),
+            dl.get_type_size_in_bits(&types, &types.i8())
+        );
+        assert_eq!(
+            Some(TypeSize::fixed(19)),
+            dl.get_type_size_in_bits(&types, &types.int(19))
+        );
+        assert_eq!(
+            Some(TypeSize::fixed(32)),
+            dl.get_type_size_in_bits(&types, &types.single())
+        );
+        assert_eq!(
+            Some(TypeSize::fixed(64)),
+            dl.get_type_size_in_bits(&types, &types.double())
+        );
+        assert_eq!(
+            Some(TypeSize::fixed(80)),
+            dl.get_type_size_in_bits(&types, &types.fp(FPType::X86_FP80))
+        );
+    }
+
+    #[test]
+    fn test_alloc_size() {
+        let dl = DataLayout::default();
+        let types = TypesBuilder::new().build();
+        assert_eq!(
+            Some(TypeSize::fixed(8)),
+            dl.get_type_alloc_size_in_bits(&types, &types.bool())
+        );
+        assert_eq!(
+            Some(TypeSize::fixed(8)),
+            dl.get_type_alloc_size_in_bits(&types, &types.i8())
+        );
+        assert_eq!(
+            Some(TypeSize::fixed(32)),
+            dl.get_type_alloc_size_in_bits(&types, &types.int(19))
+        );
+        assert_eq!(
+            Some(TypeSize::fixed(32)),
+            dl.get_type_alloc_size_in_bits(&types, &types.single())
+        );
+        assert_eq!(
+            Some(TypeSize::fixed(64)),
+            dl.get_type_alloc_size_in_bits(&types, &types.double())
+        );
+        // FP80 has no alignment information in default data layout...
+    }
+
+    #[test]
+    fn test_struct_size() {
+        let dl = DataLayout::default();
+        let types = TypesBuilder::new().build();
+        assert_eq!(
+            Some(TypeSize::fixed(0)),
+            dl.get_type_alloc_size_in_bits(
+                &types,
+                &Type::StructType {
+                    element_types: vec![],
+                    is_packed: false
+                }
+            )
+        );
+        // The i8 needs to be padded for the i32
+        assert_eq!(
+            Some(TypeSize::fixed(64)),
+            dl.get_type_alloc_size_in_bits(
+                &types,
+                &Type::StructType {
+                    element_types: vec![types.i8(), types.i32()],
+                    is_packed: false
+                }
+            )
+        );
+        // The i8 needs to be padded for the pointer
+        assert_eq!(
+            Some(TypeSize::fixed(128)),
+            dl.get_type_alloc_size_in_bits(
+                &types,
+                &Type::StructType {
+                    element_types: vec![types.i8(), types.pointer_to(types.i8())],
```

Comment:
With LLVM 15, typed pointers were removed in favour of untyped pointers. Since the type doesn't matter here, you could have a helper function along the lines of:
```rust
    fn pointer_to(types : &Types, pointee_type: TypeRef) -> TypeRef {
        #[cfg(feature = "llvm-14-or-lower")]
        {
            types.pointer_to(pointee_type)
        }
        #[cfg(feature = "llvm-15-or-greater")]
        {
            types.pointer()
        }
    }
```
or something like that, to get it to work with both typed and typeless pointers